actioncable/lib/action_cable/subscription_adapter/redis.rb

```diff
@@ -155,7 +155,7 @@ def ensure_listener_running
 
             def when_connected(&block)
               if @subscribed_client
-                block.call
+                yield
```

Comment:
I don't feel particularly strongly (don't consider this a blocker) but this is the case I don't like. IMO if we're accepting `&block`, and using `block` in another branch, then it's more readable to use `block` consistently.

Reply:
\+1

TBH, of the two such changes that were proposed, I preferred the `if block` one, for the same reason.

Performance aside, my general preference is that either a method has an explicit block parameter as an argument, and then that variable is used consistently, _or_ a method does not have an explicit argument, and then `block_given?` and `yield` are used. Mixing them feels like giving two different names to the same thing.

Whether it's _reasonable_ for us to put performance aside -- especially in a "project-wide rule" context -- is another matter, though.

Reply:
In this specific method, it has no performance impact, because `block.call` is optimized out as being strictly equivalent to `yield`.

So seems like there's some kind of consensus for not including these changes I think.

> Whether it's reasonable for us to put performance aside 

I'd say for this to matter you'd need to be in an extremely hot spot, which usually we notice via profiling. That's why I'm always so so on rubocop-performance, as most of the time it really don't matter.